be/src/storage/vectorized/merge_iterator.cpp

```diff
@@ -26,11 +26,30 @@ inline int compare_chunk(size_t key_columns, const Chunk& lhs, size_t m, const C
     return 0;
 }
 
+// MergeChunk contains a chunk for merge and an index of compared row.
+class MergeChunk {
+public:
+    MergeChunk() {}
+    explicit MergeChunk(Chunk* chunk) : _chunk(chunk) {}
+
+    size_t compared_row() const { return _compared_row; }
+
+    void advance(size_t row) { _compared_row += row; }
+
+    size_t remaining_rows() const { return _chunk->num_rows() - _compared_row; }
+
+protected:
+    friend class MaskMergeIterator;
+
+    Chunk* _chunk = nullptr;
+    uint16_t _compared_row = 0;
```

Comment:
why use uint16_t, not size_t

Reply:
use uint16_t for better heap merge performance.
I add comment and check when filling the heap.